### PR TITLE
feat(Stylelint): ✨ Initial Stylelint setup

### PR DESCRIPTION
**Pull request by xeho91**
# What is the purpose of this Pull Request?

Add an initial configuration package setup for [Stylelint].

## Type of this Pull Request

-   ✨ Implementing a new feature
-   📝 Documentation update

---

## Resources

-   [Stylelint]
-   [stylelint-order]
-   [Awesome Stylelint]

[stylelint]: https://stylelint.io/
[stylelint-order]: https://github.com/hudochenkov/stylelint-order
[awesome stylelint]: https://github.com/stylelint/awesome-stylelint

---

## TODO

-   [x] Add initial config
-   [x] Configure base plugins
-   [x] Configure with config standards _(for Sass and non-sass projects)_
-   [x] Configure `stylelint-order` plugin @acwo
-   [x] Document it _(Add a `README.md` file)_ @xeho91

### CODE REVIEW

File: packages/stylelint/source/plugins/order.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import type { Config } from "stylelint";
+
+// https://github.com/hudochenkov/stylelint-order
+const config: Partial<Config> = {
+	plugins: ["stylelint-order"],
+
+	rules: {
+		"order/order": ["custom-properties", "declarations"],
+		"order/properties-order": [
```
Comment (adchitects-co) [inline packages/stylelint/source/plugins/order.ts:9]: @acwo 

There you can continue with what you already have configured before for the company.
Let me know if you encounter any issues.

**Comment by xeho91** [comment]
@acwo I am not sure if there are any other configs/plugins you'd like to add.

You can do the research from [Awesome Stylelint](https://github.com/stylelint/awesome-stylelint)

File: packages/stylelint/source/stylelint.ts [hunk 2]
```diff
@@ -28,6 +28,32 @@ const config: Partial<Config> = {
...
+		"declaration-colon-space-after": "always",
+		"function-url-quotes": "always",
+		"selector-pseudo-element-colon-notation": "double",
+		"selector-pseudo-class-parentheses-space-inside": "never",
+		"media-feature-range-operator-space-before": "always",
+		"media-feature-range-operator-space-after": "always",
+		"media-feature-parentheses-space-inside": "never",
+		"media-feature-colon-space-before": "never",
+		"media-feature-colon-space-after": "always",
+		"scss/at-rule-no-unknown": true,
```
Comment (xeho91) [inline packages/stylelint/source/stylelint.ts:46]: @acwo Duplication. I'll resolve it and tag you where this rule is set up.

File: packages/stylelint/source/stylelint.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		 * @see https://stylelint.io/user-guide/rules/list/declaration-empty-line-before
+		 * RATIONALE: Prevent conflicting with `order` plugin
+		 */
+		"declaration-empty-line-before": undefined,
+
+		/**
+		 * @see: https://stylelint.io/user-guide/rules/list/selector-pseudo-class-no-unknown
+		 * RATIONALE: Allow using CSS Modules
+		 */
+		"selector-pseudo-class-no-unknown": [
```
Comment (xeho91) [inline packages/stylelint/source/stylelint.ts:69]: @acwo Here. I've tweaked it to allow `local` or `global`, which are used with CSS modules.

[More info about them](https://github.com/css-modules/css-modules#composing-from-global-class-names).

**Pull request merged**